pkg/solana/fees/block_history_test.go

```diff
@@ -15,82 +16,442 @@ import (
 
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
 	clientmock "github.com/smartcontractkit/chainlink-solana/pkg/solana/client/mocks"
 	cfgmock "github.com/smartcontractkit/chainlink-solana/pkg/solana/config/mocks"
 )
 
-func TestBlockHistoryEstimator(t *testing.T) {
+func TestBlockHistoryEstimator_InvalidBlockHistorySize(t *testing.T) {
+	// Helper variables for tests
 	min := uint64(10)
 	max := uint64(1000)
+	defaultPrice := uint64(100)
+	invalidDepth := uint64(0) // Invalid value to trigger default case
+	pollPeriod := 1 * time.Second
 
+	// Setup
 	rw := clientmock.NewReaderWriter(t)
 	rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
 		return rw, nil
 	})
 	cfg := cfgmock.NewConfig(t)
-	cfg.On("ComputeUnitPriceDefault").Return(uint64(100))
-	cfg.On("ComputeUnitPriceMin").Return(min)
-	cfg.On("ComputeUnitPriceMax").Return(max)
-	cfg.On("BlockHistoryPollPeriod").Return(100 * time.Millisecond)
-	lgr, logs := logger.TestObserved(t, zapcore.DebugLevel)
+	setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, invalidDepth)
+	lgr, _ := logger.TestObserved(t, zapcore.DebugLevel)
 	ctx := tests.Context(t)
+	estimator := initializeEstimator(t, rwLoader, cfg, lgr, ctx)
 
-	// file contains legacy + v0 transactions
-	testBlockData, err := os.ReadFile("./blockdata.json")
-	require.NoError(t, err)
-	blockRes := &rpc.GetBlockResult{}
-	require.NoError(t, json.Unmarshal(testBlockData, blockRes))
+	// Call calculatePrice and expect an error due to invalid BlockHistorySize
+	err := estimator.calculatePrice(ctx)
+	require.Error(t, err, "Expected error for invalid block history size")
+	assert.Equal(t, "invalid block history depth: 0", err.Error(), "Unexpected error message for invalid block history size")
+
+	// Ensure the compute unit price remains at the default value
+	cfg.On("ComputeUnitPriceMax").Return(max)
+	assert.Equal(t, defaultPrice, estimator.BaseComputeUnitPrice(), "Price should remain at default for invalid block history size")
+}
+
+func TestBlockHistoryEstimator_LatestBlock(t *testing.T) {
+	// Helper variables for tests
+	min := uint64(10)
+	max := uint64(100_000)
+	defaultPrice := uint64(100)
+	depth := uint64(1) // 1 is LatestBlockEstimator
+	pollPeriod := 100 * time.Millisecond
+
+	// Grabbing last block of multiple_blocks file to use as latest block
+	testBlocks := readMultipleBlocksFromFile(t, "./multiple_blocks_data.json")
+	lastBlock := testBlocks[len(testBlocks)-1]
+	lastBlockFeeData, _ := ParseBlock(lastBlock)
+	lastBlockMedianPrice, _ := mathutil.Median(lastBlockFeeData.Prices...)
+
+	t.Run("Successful Estimation", func(t *testing.T) {
+		// Setup
+		rw := clientmock.NewReaderWriter(t)
+		rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+			return rw, nil
+		})
+		cfg := cfgmock.NewConfig(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, depth)
+		rw.On("GetLatestBlock", mock.Anything).Return(lastBlock, nil)
```

Comment:
To reduce some of the duplication here, could we move this mock ReaderWriter to the parent test with the mock for `rw.On("GetLatestBlock", mock.Anything).Return(lastBlock, nil)`. Seems like this is the same across a few tests. It's ok if you mock new ReaderWriters for the RPC failure tests further down though